app/controllers/users_controller.rb

```diff
@@ -0,0 +1,27 @@
+class UsersController < ApplicationController
+  def create
+  	user = User.find_by(email: params[:emails]["0"][:value])
+  	if !user
+  		# Strip trailing "?sz=50" from Google Plus image URL
+  		image_url = params[:image][:url]
+  		image_url = image_url[0..image_url.index('?') - 1]
+
+  		user = User.create(first_name:params[:name][:givenName], 
+  						last_name:params[:name][:familyName], 
+  						email:params[:emails]["0"][:value], 
+  						image:image_url)
+  	end
+    log_in user
+  	render text: "success"
+  end
+
+  def update
+  	params.permit!
```

Comment:
again ... this is trouble.  I could conceivably create my user with `has_root_access_pwned: true` set in the hash.